step-03/lib/search_bar.dart

```diff
@@ -1,56 +1 @@
-// Copyright 2018 The Flutter team. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
 
-import 'package:flutter/cupertino.dart';
-import 'package:flutter/widgets.dart';
-import 'styles.dart';
-
-class SearchBar extends StatelessWidget {
-  const SearchBar({
-    @required this.controller,
-    @required this.focusNode,
-  });
-
-  final TextEditingController controller;
-  final FocusNode focusNode;
-
-  @override
-  Widget build(BuildContext context) {
-    return DecoratedBox(
-      decoration: BoxDecoration(
-        color: Styles.searchBackground,
-        borderRadius: BorderRadius.circular(10),
-      ),
-      child: Padding(
-        padding: const EdgeInsets.symmetric(
-          horizontal: 4,
-          vertical: 8,
-        ),
-        child: Row(
-          children: [
-            const Icon(
-              CupertinoIcons.search,
-              color: Styles.searchIconColor,
-            ),
-            Expanded(
-              child: CupertinoTextField(
-                controller: controller,
-                focusNode: focusNode,
-                style: Styles.searchText,
-                cursorColor: Styles.searchCursorColor,
-              ),
-            ),
-            GestureDetector(
```

Comment:
I forget what we said in the other commit. Could we just use the clear button in the CupertinoTextField?

Reply:
The clear button in CupertinoTextField doesn't work with the dark background this search text field has. This code is lifted from @redbrogdon's Veggie Seasons sample, so maybe he can give more colour.

Reply:
Doh! Can you open an issue on the main repo?

Reply:
Whups, sorry I missed this. I see you have already pushed through a PR for this. Any ideas on when it will land?